src/library/itunes/itunesmacosimporter.mm

```diff
@@ -81,10 +81,21 @@ void importMediaItems(NSArray<ITLibMediaItem*>* items) {
     }
 
     void appendPlaylistTree(TreeItem& item, int playlistId = -1) {
-        for (auto it = m_playlistChildsByDbId.find(playlistId);
-                it != m_playlistChildsByDbId.end() && it.key() == playlistId;
-                ++it) {
-            int childId = it.value();
+        // Find bounds of the child ID range in the multimap
+        auto childIdsBegin = m_playlistChildsByDbId.constFind(playlistId);
+        auto childIdsEnd = childIdsBegin;
+        while (childIdsEnd != m_playlistChildsByDbId.constEnd() &&
+                childIdsEnd.key() == playlistId) {
+            ++childIdsEnd;
+        }
+
+        // QMultiMap stores items with same key in reverse insertion order,
+        // therefore iterate in reverse to get the insertion order again.
+        // See the docs: https://doc.qt.io/qt-5/qmultimap.html#details
```

Comment:
This is unfortunately a Qt5 implementation detail. With Qt6 it depends on the used std:: implementation. 
Does the code rely on it? Than we may consider to force it by `QMultiMap<int, std::vector<int>>`

Reply:
That kind of sidesteps the point of using a multimap, in that case we could just use a regular map or hash (which is fine, my thinking was that the multimap is likely more efficient when working on large playlist trees)

Reply:
The naive implementation would be to just get the equal range as a `QList` or `std::vector` and sort it, which is O(n log n), rather than O(n). But that's maybe still acceptable, probably much clearer and doesn't rely on unnecessarily many details. Wdyt?

Reply:
How many values could there be present for a single key? IIU the code correctly, there are many keys but few values. In that case we should optimize for constant overhead instead. Also, if using a `std::multimap` has advantages, lets just use it directly. It's not like we need to pass it to any other Qt Apis. 
`std::multimap::equal_range` seems perfect for our use case. We also don't need to do the reversing because the standard explicitly dictates that the order of the elements in the range is in insertion order unless specified otherwise explicitly).

Reply:
That very much depends on the user's playlist structure. If a user has a deep tree of folders with few playlists, then yes, but if the user prefers a relatively flat hierarchy there will be lots of values for few keys. `std::multimap` sounds promising, I'll look into that.

Reply:
```cpp
// renamed from m_playlistChildsByDbId
std::pair allChildrenOfParentPlaylist = m_playlistDbIdsByParentDbId.equal_range(playlistId);

std::for_each(allChildrenOfParentPlaylist.first, allChildrenOfParentPlaylist.second, [](std::pair kv) {const int childId = kv.second; ...});

```